pkg/controllers/multicluster/utils.go

```diff
@@ -40,6 +41,28 @@ const (
 	ValueEndpointSliceManagedBy = "aws-cloud-map-mcs-controller-for-k8s"
 )
 
+func AddressTypeToIPType(addressType discovery.AddressType) (model.IPType, error) {
+	switch addressType {
+	case discovery.AddressTypeIPv6:
+		return model.IPV6Type, nil
+	case discovery.AddressTypeIPv4:
+		return model.IPV4Type, nil
+	default:
+		return "", fmt.Errorf("unsupported address type %s", addressType)
+	}
+}
+
+func IPTypeToAddressType(ipType model.IPType) discovery.AddressType {
```

Comment:
Do we need a model type for IP? Can we just use discovery.AddressType?

Reply:
Good point; I'm changing it over to discovery.AddressType across the codebase.